### PR TITLE
Issues/2397

### LINKED ISSUE #2397: Add feature to sort donations on basis of gateway
Go to Reports -> Donation Methods, On this page you can see donation counts per gateways. This will be good if we will create gateway filter link for each gateway which we help to admin to watch list of donations per gateway.
image

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

THIS PR WILL RESOLVE #2397 

## How Has This Been Tested?

Manually apply gateway sorting with form_id

## Screenshots (jpeg or gifs if applicable):
image
image
image

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/payments/functions.php [hunk 1]
```diff
@@ -419,177 +419,29 @@ function give_undo_donation( $payment_id ) {
  * @return object $stats Contains the number of payments per payment status.
  */
 function give_count_payments( $args = array() ) {
```
Comment (ravinderk) [inline includes/payments/functions.php]: @DevinWalker I update this function and removed some unused filter also, let me know what do you think

Comment (ravinderk) [reply includes/payments/functions.php]: Let me know if you want to add backward compatibility for `give_count_payments_where` and `give_count_payments_join`

**Comment by raftaar1191** [comment]
@ravinderk have a Question for you


We can order table with donation from id, date etc but what I see is that I am not able to use any of the **filter**  

Is it normal?

image

**Comment by ravinderk** [comment]
@DevinWalker and @mehul0810 can you confirm sorting is working fine for you

**Comment by DevinWalker** [comment]
@ravinderk sorting what?

**Comment by ravinderk** [comment]
@DevinWalker Sorting on donation listing page with filters

**Comment by DevinWalker** [comment]
@ravinderk looks like sorting is working but paging is broken https://goo.gl/qUqutJ

Can you fix that ASAP?

**Pull request merged**